loading.bs

```diff
@@ -0,0 +1,949 @@
+
+A [=request=] has an associated <dfn for="request">stashed exchange</dfn>, which
+is null or an [=exchange=].
+
+<h3 algorithm id="mp-request-clone">Request clone</h3>
+
+Rewrite [=request/clone|clone a request=] to run these steps:
+
+<ol>
+ <li>Let |newRequest| be a copy of |request|, except for its [=request/body=]
+  <ins>and [=request/stashed exchange=]</ins>.
+
+ <li>If |request|'s [=request/body=] is non-null, set |newRequest|'s
+  [=request/body=] to the result of [=body/cloning=] |request|'s
+  [=request/body=].
+
+ <li><ins>If |request|'s [=request/stashed exchange=] is non-null, set
+  |newRequest|'s [=request/stashed exchange=] to an exchange whose
+  [=exchange/request=] is the [=request/clone=] of |request|'s [=request/stashed
+  exchange=]'s [=exchange/request=] and whose [=exchange/response=] is the
+  [=response/clone=] of |request|'s [=request/stashed exchange=]'s
+  [=exchange/response=].</ins>
+
+ <li>Return |newRequest|.
+</ol>
+
+<h3 algorithm id="mp-response-date">Response date</h3>
+
+A [=response=] |response|'s <dfn for="response">date</dfn> is the result of:
+
+1. Let |date| be the result of [=extracting header list values=] given `` `Date`
+    `` and |response|'s [=response/header list=].
+1. If |date| is a failure, return the point in time of the beginning of the
+    universe.
+1. Return the point in time represented by |date|, as interpreted for the [=Date
+    header field=].
+
+<h3 algorithm id="mp-http-fetch">Monkeypatch HTTP fetch</h3>
+
+In [=HTTP fetch=], before
+
+> 5. If |actualResponse|’s [=response/status=] is a [=redirect status=], then:
+>     ...
+
+add the following steps:
+
+5. If the [=signed exchange version=] of |actualResponse| is:
+
+    <dl class="switch">
+    : undefined
+    :: Do nothing.
+
+    : `"b1"`
+    ::
+        1. Let |parsedExchange| be the result of [=parsing a b1 signed
+            exchange=] from |actualResponse| in the context of |request|'s
+            [=request/client=].
+        1. If |parsedExchange| is a failure, return a [=network error=].
+        1. Set |actualResponse|'s [=status=] to `303`.
+        1. [=header list/Set=] |actualResponse|'s `` `Location` `` header to
+            the [=UTF-8 encoding=] of the [=URL serializer|serialization=]
+            of |parsedExchange|'s [=exchange/request=]'s [=request/url=].
+        1. Set |request|'s [=request/stashed exchange=] to |parsedExchange|.
+
+    : Anything else
+    :: Set |actualResponse| to a [=network error=].
+
+    </dl>
+
+    Note: The final [[draft-yasskin-http-origin-signed-responses]] will use a
+    version of `` `1` ``, but this specification tracks what's actually
+    implemented in browsers, which still uses draft versions.
+
+<h3 algorithm id="mp-http-network-or-cache-fetch">Monkeypatch HTTP-network-or-cache fetch</h3>
+
+In <a spec="fetch">HTTP-network-or-cache fetch</a>, after
+
+> 5.19. If |httpRequest|’s [=request/cache mode=] is neither "`no-store`" nor
```

Comment:
Right now, I'm ignoring all of the [cache modes](https://fetch.spec.whatwg.org/#concept-request-cache-mode) in deciding to accept a signed exchange's content. They may be useful in controlling how to fetch the `.sxg` itself, and applying them to the content would probably just have the effect of forcing that to the network, which would defeat the point.

Does that sound right to y'all, or do you see any of the modes I should apply here?

Reply:
This sg to me.